Extract setup in lib

pgrange opened the pull request:
With bash_unit 1.5.0 the current working directory is changed by bash_unit before the test file is loaded.

This way, you can source a lib file containing setup functions without taking care of where you code is run from.

Constraint/advantage of this commit: one can know run all test files and not only test_main.sh

```
./bash_unit test*
```

Pamplemousse commented:
I see two problems with the current PR:

  * `README.md` should be updated with a proper command to run tests
  * all the tests related to the parameter values, in `tests/ssh_parameters/test_*` are not run anymore

---

Trying to correct this, I am not sure if it fixes the lack of elegance of the current solution.

In particular, this statement is not always true:

> This way, you can source a lib file containing setup functions without taking care of where you code is run from.

For example, when tests files are organized in different sub-folders (as it is the case here with `tests/ssh_parameters/`), it's still necessary to "keep track" of the structure when sourcing the lib (setup and teardown): `source ../lib_test`.

pgrange commented:
Sorry, didn't see tests/ssh_parameters/ subdirectory. I updated them to include ```../lib_test.sh```

I updated the README to run all tests, even in tests sub-directories.

You are right, it is necessary to keep track of the directory structure to source files in bash. I don't know other ways to do so.
